server/etcdserver/server.go

```diff
@@ -1081,9 +1084,10 @@ func (s *EtcdServer) applyEntries(ep *etcdProgress, apply *apply) {
 }
 
 func (s *EtcdServer) triggerSnapshot(ep *etcdProgress) {
-	if ep.appliedi-ep.snapi <= s.Cfg.SnapshotCount {
+	if !s.forceSnapshot && ep.appliedi-ep.snapi <= s.Cfg.SnapshotCount {
 		return
 	}
+	s.forceSnapshot = false
```

Comment:
It might be worth to log the reason for snapshot being created when it was forced.

Reply:
(probably expand the line 'triggering snapshot') .

Reply:
Done